rust/crd/src/lib.rs

```diff
@@ -818,6 +821,55 @@ impl DruidCluster {
 
         fragment::validate(rolegroup_config).context(FragmentValidationFailureSnafu)
     }
+
+    pub fn pod_overrides_for_role(&self, role: &DruidRole) -> &PodTemplateSpec {
+        match role {
+            DruidRole::Broker => &self.spec.brokers.config.pod_overrides,
+            DruidRole::Coordinator => &self.spec.coordinators.config.pod_overrides,
+            DruidRole::Historical => &self.spec.historicals.config.pod_overrides,
+            DruidRole::MiddleManager => &self.spec.middle_managers.config.pod_overrides,
+            DruidRole::Router => &self.spec.routers.config.pod_overrides,
+        }
+    }
+
+    pub fn pod_overrides_for_role_group(
+        &self,
+        role: &DruidRole,
+        role_group: &str,
+    ) -> Option<&PodTemplateSpec> {
+        match role {
+            DruidRole::Broker => self
+                .spec
+                .brokers
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+            DruidRole::Coordinator => self
+                .spec
+                .coordinators
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+            DruidRole::Historical => self
+                .spec
+                .historicals
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+            DruidRole::MiddleManager => self
+                .spec
+                .middle_managers
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+            DruidRole::Router => self
+                .spec
+                .routers
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+        }
```

Comment:
Could just extract the role_groups in the match and do the get() outside once?

Reply:
Nope :/
```
error[E0308]: `match` arms have incompatible types
   --> rust/crd/src/lib.rs:842:39
    |
840 | /         match role {
841 | |             DruidRole::Broker => self.spec.brokers.role_groups,
    | |                                  ----------------------------- this is found to be of type `HashMap<std::string::String, RoleGroup<BrokerConfigFragment>>`
842 | |             DruidRole::Coordinator => self.spec.coordinators.role_groups,
    | |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `BrokerConfigFragment`, found struct `CoordinatorConfigFragment`
843 | |             DruidRole::Historical => self.spec.historicals.role_groups,
844 | |             DruidRole::MiddleManager => self.spec.middle_managers.role_groups,
845 | |             DruidRole::Router => self.spec.routers.role_groups,
846 | |         }
    | |_________- `match` arms have incompatible types
    |
    = note: expected struct `HashMap<_, RoleGroup<BrokerConfigFragment>>`
               found struct `HashMap<_, RoleGroup<CoordinatorConfigFragment>>
````

Reply:
ok resolved! :D